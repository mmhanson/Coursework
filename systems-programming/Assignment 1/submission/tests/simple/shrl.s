main:
	movl	$21, %eax
	shrl	%eax
	printr	%eax
	shrl	%eax
	printr	%eax
	shrl	%eax
	printr	%eax
	shrl	%eax
	printr	%eax
	shrl	%eax
	printr	%eax
	movl	$-1, %ebx
	shrl	%ebx
	printr	%ebx
	movl	$-32768, %ecx
	shrl	%ecx
	printr	%ecx
