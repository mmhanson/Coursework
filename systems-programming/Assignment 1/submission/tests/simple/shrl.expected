10 (0xa)
5 (0x5)
2 (0x2)
1 (0x1)
0 (0x0)
2147483647 (0x7fffffff)
2147467264 (0x7fffc000)
